lib/ansible/modules/cloud/vmware/vmware_host_hyperthreading.py

```diff
@@ -0,0 +1,246 @@
+
+            if enable_hyperthreading:
+                results['result'][host.name]['desired_state'] = "enabled"
+                # Don't do anything if Hyperthreading is already enabled
+                if hyperthreading_info.config and hyperthreading_info.active:
+                    results['result'][host.name]['previous_state'] = "enabled"
+                    results['result'][host.name]['current_state'] = "enabled"
+                    results['result'][host.name]['msg'] = "Hyperthreading is already enabled and active for host '%s'" % \
+                                                          host.name
+                elif hyperthreading_info.config and hyperthreading_info.active is False:
+                    changed = True
+                    results['result'][host.name]['previous_state'] = "enabled"
+                    results['result'][host.name]['current_state'] = "enabled"
+                    results['result'][host.name]['msg'] = ("Hyperthreading is already enabled for host '%s', but not active."
+                                                           " A reboot is required!" % host.name)
+                # Enable Hyperthreading
+                else:
+                    # Check if Hyperthreading is available
+                    if hyperthreading_info.available:
+                        if not self.module.check_mode:
+                            try:
+                                host.configManager.cpuScheduler.EnableHyperThreading()
+                                changed = True
+                                results['result'][host.name]['changed'] = True
+                                results['result'][host.name]['previous_state'] = "disabled"
+                                results['result'][host.name]['current_state'] = "enabled"
+                                results['result'][host.name]['msg'] = "Hyperthreading enabled for host '%s'" % host.name
+                            except vmodl.fault.NotSupported as not_supported:
+                                # This should never happen since Hyperthreading is available
+                                self.module.fail_json(msg="Failed to enable Hyperthreading for host '%s' : %s" %
+                                                      (host.name, to_native(not_supported.msg)))
+                            except (vmodl.RuntimeFault, vmodl.MethodFault) as runtime_fault:
+                                self.module.fail_json(msg="Failed to enable Hyperthreading for host '%s' due to : %s" %
+                                                      (host.name, to_native(runtime_fault.msg)))
+                        else:
+                            changed = True
+                            results['result'][host.name]['changed'] = True
+                            results['result'][host.name]['previous_state'] = "disabled"
+                            results['result'][host.name]['current_state'] = "disabled"
```

Comment:
Same as with the other module: I suggest setting the current/desired state to what it would actually become, including the correct message. Check mode should represent what will be the outcome of the change as if it were for real, except not actually make the change.